docs/doxygen/porting.dox

```diff
@@ -0,0 +1,103 @@
+ - @ref LogError
+ - @ref LogWarn
+ - @ref LogInfo
+ - @ref LogDebug
+
+@section sntp_porting_transport UDP Transport Interface
+@brief The coreSNTP library exposes a UDP transport interface API that must be implemented
+in order to send and receive SNTP packets over the network.
+
+@see [UDP Transport Interface](@ref UdpTransportInterface_t)
+
+@note For security against unwanted server response packets, it is RECOMMENDED that the UDP socket that is used
+for implementing the UDP transport interface functions of performing network I/O is kept open ONLY during the
+duration of a request-response SNTP communication cycle instead of keeping the socket always open. One way to
+achieve this is to open a new UDP socket before calling @ref Sntp_SendTimeRequest API and closing the socket after
+receiving server response (or performing appropriate wait for response) with the @ref Sntp_ReceiveTimeResponse API.
+
+A port must implement functions corresponding to the following functions pointers:
+ - [UDP Transport Send](@ref UdpTransportSendTo_t): A function to send bytes on the network to the time server
+ passed to it. It is RECOMMENDED that the function is implemented with non-blocking in behavior, so
+ the total block time for send operation can be passed as runtime parameter to @ref Sntp_SendTimeRequest
+ API.
+
+ - [UDP Transport Receive](@ref UdpTransportRecvFrom_t): A function to receive bytes over the network from the
+ time server passed to it. It is RECOMMENDED that the function is implemented with non-blocking in behavior, so
+ the total block time for receive operation can be passed as runtime parameter to @ref Sntp_ReceiveTimeResponse
+ API.
+
+The above two functions take in a pointer to a @ref NetworkContext_t, the typename of a
+`struct NetworkContext`. The NetworkContext struct must also be defined by the port, and
+ought to contain any information necessary to send and receive data with the @ref UdpTransportSendTo_t
+and @ref UdpTransportRecvFrom_t implementations, respectively:
+@code
+// Example definition of NetworkContext
+struct NetworkContext {
+    // Fields necessary for the transport implementations, e.g. a UDP socket descriptor.
+};
+@endcode
+
+@section sntp_porting_gettime Get Time Function
+@brief The coreSNTP library relies on this function to obtain time from system for the operation of tracking timeout
+durations as well as generating time request packet required by the SNTP protocol.
+
+@see @ref SntpGetTime_t
+
+In the scenario that device platform does not have real-world time information (due to device boot-up for example), it is
+acceptable for this function to provide the system-time information that does not match real-world time, because once a
+time information is received in SNTP communication with a time server, the system time can be corrected to match real-time.
+Refer to the next section on how the interface exposed by coreSNTP library to correct system time.
```

Comment:
```suggestion
If the device does not have real-world time information (on device boot-up for example), it is
acceptable for this function to provide the system-time that does not match the real-world time, because once a
time information is received from a time server, the system time can be corrected to match the real-world time.
Refer to the next section on how to correct the system time.
```